### ISSUE TITLE
default value is useless without isValid call, which defaults to true for everything

### ISSUE THREAD

**Issue by AVVS** [jonschlinkert/get-value#15]
Hi, I've noticed that defaultValue in the case like this:

get(options, 'property', false) would always return the value of property and would never return false

was this intentional to make isValid always return true? I cant imagine a case where one would want to define a default value and then never be able to get it with the current isValid implementation

Sorry for the remark if that was done by design, but if not - I imagine it would be great to update the lib

**Comment by doowb** [comment]
`default` should be returned if the `value` of the requested `key` (or `path`) is not on the target (`options` in this case) or if a user provided `isValid` function returns `false`.

So the case that you're showing would return the `value` of `property` if `options.property` is set, otherwise, it will return `false`. Looking closely at the code, there are `if(key in target)` statements before calling `if(isValid(key, target))` which means that the property has to be on the `target`.

In other words, the following 2 statements should be equivalent:

```js
const a = 'property' in options ? options.foo : false;
const b = get(options, 'property', false);
```

I hope this helps. I'm going to close the issue, but if you have more questions about this, feel free to comment on this issue.

**Comment by AVVS** [comment]
thanks, indeed missed the `in path` check
